ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -0,0 +1,298 @@
+  );
+
+  const subjectMetadata: { [key: string]: any } = useSelector(
+    getConnectedSitesList,
+  );
+  const connectedSubjectsMetadata = subjectMetadata[activeTabOrigin];
+
+  const connectedNetworks = useSelector((state) =>
+    getPermittedChainsForSelectedTab(state, activeTabOrigin),
+  ) as string[];
+
+  const permittedAccountsByOrigin = useSelector(
+    getPermittedAccountsByOrigin,
+  ) as PermittedAccountsByOrigin;
+  const networksList = useSelector(getNonTestNetworks);
+  const testNetworks = useSelector(getTestNetworks);
+  const combinedNetworks = [...networksList, ...testNetworks];
+
+  const connectedAccounts = useSelector((state) =>
+    getOrderedConnectedAccountsForConnectedDapp(state, activeTabOrigin),
+  ) as AccountType[];
+
+  const subjects = useSelector(getPermissionSubjects);
+  const grantedNetworks = combinedNetworks.filter(
+    (net: { chainId: any }) => connectedNetworks.indexOf(net.chainId) !== -1,
+  );
+
+  const hostName = getURLHost(securedOrigin);
+  const currentTabHasNoAccounts =
+    !permittedAccountsByOrigin[activeTabOrigin]?.length;
+
+  let tabToConnect: { origin: string | null } = { origin: null };
+  if (activeTabOrigin && currentTabHasNoAccounts && !openMetaMaskTabs[id]) {
+    tabToConnect = {
+      origin: activeTabOrigin,
+    };
+  }
+
+  const requestAccountsAndChainPermissions = async () => {
+    if (tabToConnect.origin) {
+      // Ensure origin is not null
+      const requestId = await dispatch(
+        requestAccountsAndChainPermissionsWithId(tabToConnect.origin),
+      );
+      history.push(`${CONNECT_ROUTE}/${requestId}`);
+    } else {
+      console.error('Tab origin is null.');
+    }
+  };
+
+  const disconnectAllAccounts = () => {
+    const subject = (subjects as SubjectsType)[activeTabOrigin];
+
+    if (subject) {
+      const permissionMethodNames = Object.values(subject.permissions).map(
+        ({ parentCapability }: { parentCapability: string }) =>
+          parentCapability,
+      ) as string[];
+      if (permissionMethodNames.length > 0) {
+        const permissionsRecord: Record<string, string[]> = {
+          [activeTabOrigin]: permissionMethodNames,
+        };
+
+        dispatch(
+          removePermissionsFor(
+            permissionsRecord as Record<string, NonEmptyArray<string>>,
+          ),
+        );
+      }
+    }
+  };
+
+  return (
+    <Page
+      data-testid="connections-page"
+      className="main-container connections-page"
+    >
+      <>
+        <Header
+          backgroundColor={BackgroundColor.backgroundDefault}
+          startAccessory={
+            <ButtonIcon
+              ariaLabel={t('back')}
+              iconName={IconName.ArrowLeft}
+              className="connections-header__start-accessory"
+              size={ButtonIconSize.Sm}
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              onClick={() => (history as any).goBack()}
```

Comment:
Do we know what type this could be?  `Window.history`?  Does this need to be variable or do we know where this should go every time?

Reply:
This goes back to the page where it was previously